src/jit/liveness.cpp

```diff
@@ -1687,204 +1687,208 @@ void Compiler::fgComputeLifeCall(VARSET_TP& life, GenTreeCall* call)
-                lclVarNode->gtFlags &= ~GTF_VAR_DEATH;
-                return false;
+                VarSetOps::RemoveElemD(this, life, varIndex);
             }
-
 #ifdef DEBUG
             if (verbose && 0)
             {
-                printf("Ref V%02u,T%02u] at ", lclNum, varIndex);
+                printf("Def V%02u,T%02u at ", node->gtLclNum, varIndex);
                 printTreeID(node);
-                printf(" life %s -> %s\n", VarSetOps::ToString(this, life),
-                       VarSetOps::ToString(this, VarSetOps::AddElem(this, life, varIndex)));
+                printf(" life %s -> %s\n",
+                       VarSetOps::ToString(this,
+                                           VarSetOps::Union(this, life, VarSetOps::MakeSingleton(this, varIndex))),
+                       VarSetOps::ToString(this, life));
             }
 #endif // DEBUG
+        }
+    }
+    else
+    {
+        /* Dead assignment to the variable */
+        node->gtFlags |= GTF_VAR_DEATH;
 
-            // The variable is being used, and it is not currently live.
-            // So the variable is just coming to life
-            lclVarNode->gtFlags |= GTF_VAR_DEATH;
-            VarSetOps::AddElemD(this, life, varIndex);
+        if (!opts.MinOpts())
+        {
+            // keepAliveVars always stay alive
+            noway_assert(!VarSetOps::IsMember(this, keepAliveVars, varIndex));
 
-            // Record interference with other live variables
-            fgMarkIntf(life, varIndex);
+            // Do not consider this store dead if the target local variable represents
+            // a promoted struct field of an address exposed local or if the address
+            // of the variable has been exposed. Improved alias analysis could allow
+            // stores to these sorts of variables to be removed at the cost of compile
+            // time.
+            return !varDsc.lvAddrExposed && !(varDsc.lvIsStructField && lvaTable[varDsc.lvParentLcl].lvAddrExposed);
         }
     }
-    // Note that promoted implies not tracked (i.e. only the fields are tracked).
-    else if (varTypeIsStruct(varDsc->lvType))
-    {
-        noway_assert(!varDsc->lvTracked);
 
-        lvaPromotionType promotionType = lvaGetPromotionType(varDsc);
+    return false;
+}
 
-        if (promotionType != PROMOTION_TYPE_NONE)
-        {
-            VarSetOps::AssignNoCopy(this, varBit, VarSetOps::MakeEmpty(this));
+void Compiler::fgComputeLifeUntrackedLocal(
+    VARSET_TP& life, VARSET_VALARG_TP keepAliveVars, LclVarDsc& varDsc, GenTreeLclVarCommon* lclVarNode, GenTree* node)
+{
+    assert(lclVarNode != nullptr);
+    assert(node != nullptr);
 
-            for (unsigned i = varDsc->lvFieldLclStart; i < varDsc->lvFieldLclStart + varDsc->lvFieldCnt; ++i)
-            {
+    if (!varTypeIsStruct(varDsc.lvType) || (lvaGetPromotionType(&varDsc) == PROMOTION_TYPE_NONE))
+    {
+        return;
+    }
+
+    VARSET_TP varBit(VarSetOps::MakeEmpty(this));
```

Comment:
Again, I know this is pre-existing code, but `varBit` seems to be a strange name for a set that is likely to have more than one member. I would probably rename this `fieldVars` or something, but IMO it should at least be plural.